src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/PSRecoveryServicesClient/PSRecoveryServicesClient.cs

```diff
@@ -259,6 +261,44 @@ private SiteRecoveryManagementClient GetSiteRecoveryClient()
 
             return siteRecoveryClient;
         }
+
+        /// <summary>
+        /// Gets Vault Upgrade client.
+        /// </summary>
+        /// <param name="location">Resource location.</param>
+        /// <param name="resourceName">Resource Name.</param>
+        /// <param name="resourceType">Resource Type.</param>
+        /// <returns>Vault Upgrade Management client.</returns>
+        private RecoveryServicesVaultUpgradeManagementClient GetVaultUpgradeClient(
+            string location,
+            string resourceName,
+            string resourceType)
+        {
+            string resourceNamespace = Environment.GetEnvironmentVariable("RESOURCE_NAMESPACE");
```

Comment:
Actually this has been added for internal testing only. 

But as you said it's quite possible that because of customer's environment, this gets used. Instead of the prod values. One check I can put here is "the value should be either of the two allowed values(WABackupBVTD2/HRMBvtd2ToDogfood)". The chances that he would be using the same value, unless intentionally, is very less.